### PR TITLE
HZN-1333: "Change Icon" In Topology Map Fails

### PR DESCRIPTION
**Pull request by patrick-schweizer**
* JIRA: https://issues.opennms.org/browse/HZN-1333

Our [continuous integration system](http://bamboo.internal.opennms.com:8085) will test and verify your changes.

### CODE REVIEW

**Comment by mvrueden** [comment]
At first glance this looks good to me. However this has an impact on all Vaadin applications not only the Topology Map, I'd like to test it out manually before merging, though. I'll come back to you, as soon as I had the chance.

**Comment by mvrueden** [comment]
I tested it and LGTM as well. 
We should add a test for changing the icon though.

**Comment by patrick-schweizer** [comment]
@mvruden I added a test that verifies that the icon can be changed.

File: smoke-test/src/test/java/org/opennms/smoketest/topo/GraphMLTopologyIT.java [hunk 1]
```diff
@@ -238,9 +238,25 @@ public void verifyCanFilterByCategory() throws IOException, InterruptedException
...
+        topologyUIPage.findVertex("North Region").contextMenu().click("Navigate To", "Markets (North Region)");
+        String label = "North 1";
+
+        // open icon select popup
+        topologyUIPage.findVertex(label).select();
+        topologyUIPage.findVertex(label).contextMenu().click("Change Icon");
+
+        // choose new icon
+        String newIconName = "microwave_backhaul_1";
+        findElementByXpath("//*[name()='title' and text()='"+newIconName+"']/../*[name()='rect']").click();
```
Comment (mvrueden) [inline smoke-test/src/test/java/org/opennms/smoketest/topo/GraphMLTopologyIT.java]: Can we move this to a `setIcon(...)` method?

Comment (patrick-schweizer) [reply smoke-test/src/test/java/org/opennms/smoketest/topo/GraphMLTopologyIT.java]: to make sure: you want me to put the whole process of choosing an icon in the method?
That would be:
* click on an Element
* open the context menu via right click
* open the change icon dialog
* select an icon
* close the icon dialog

Comment (mvrueden) [reply smoke-test/src/test/java/org/opennms/smoketest/topo/GraphMLTopologyIT.java]: Hmm sorry, the comment was in state `pending`, but I wrote it 4 days ago, is it visible to you now?

Comment (patrick-schweizer) [reply smoke-test/src/test/java/org/opennms/smoketest/topo/GraphMLTopologyIT.java]: now it is :-) 
the code snipped you posted does steps 1-3 but I assume we also want step 4. I'll wrap it in a method.

Comment (mvrueden) [reply smoke-test/src/test/java/org/opennms/smoketest/topo/GraphMLTopologyIT.java]: Yeah, makes sense (-:

**Comment by RangerRick** [comment]
Yay!  Looks good!

**Pull request merged**